### ISSUE TITLE
"java.lang.IllegalStateException: Connection pool shut down" returned when refresh endpoint is called

### ISSUE THREAD

**Issue by AlexandraMDan** [spring-cloud/spring-cloud-config#918]
I have created a Spring Cloud Config Server using Spring Boot 1.5.10 and Spring Cloud version Edgware.SR2 and I have an issue after calling refresh endpoint. My application is using RestTemplate to make an HTTP request and after calling refresh endpoint the following exception appears: 

> java.lang.IllegalStateException: Connection pool shut down
	at org.apache.http.util.Asserts.check(Asserts.java:34) ~[httpcore-4.4.8.jar:4.4.8]
	at org.apache.http.pool.AbstractConnPool.lease(AbstractConnPool.java:191) ~[httpcore-4.4.8.jar:4.4.8]
	at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.requestConnection(PoolingHttpClientConnectionManager.java:257) ~[httpclient-4.5.3.jar:4.5.3]
	at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:176) ~[httpclient-4.5.3.jar:4.5.3]
	at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:185) ~[httpclient-4.5.3.jar:4.5.3]
	at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89) ~[httpclient-4.5.3.jar:4.5.3]
	at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:111) ~[httpclient-4.5.3.jar:4.5.3]
	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185) ~[httpclient-4.5.3.jar:4.5.3]
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83) ~[httpclient-4.5.3.jar:4.5.3]
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:56) ~[httpclient-4.5.3.jar:4.5.3]
	at org.springframework.http.client.HttpComponentsClientHttpRequest.executeInternal(HttpComponentsClientHttpRequest.java:89) ~[spring-web-4.3.13.RELEASE.jar:4.3.13.RELEASE]
	at org.springframework.http.client.AbstractBufferingClientHttpRequest.executeInternal(AbstractBufferingClientHttpRequest.java:48) ~[spring-web-4.3.13.RELEASE.jar:4.3.13.RELEASE]
	at org.springframework.http.client.AbstractClientHttpRequest.execute(AbstractClientHttpRequest.java:53) ~[spring-web-4.3.13.RELEASE.jar:4.3.13.RELEASE]
	at org.springframework.web.client.RestTemplate.doExecute(RestTemplate.java:659) ~[spring-web-4.3.13.RELEASE.jar:4.3.13.RELEASE]
	at org.springframework.web.client.RestTemplate.execute(RestTemplate.java:628) ~[spring-web-4.3.13.RELEASE.jar:4.3.13.RELEASE]
	at org.springframework.web.client.RestTemplate.getForObject(RestTemplate.java:302) ~[spring-web-4.3.13.RELEASE.jar:4.3.13.RELEASE]

If I am using Edgware.RELEASE version for spring cloud, the application works fine. I have noticed that the issue is caused by **spring-cloud-commons-dependencies** which was changed - from 1.3.0.RELEASE version in Edgware.RELEASE to 1.3.2.RELEASE in Edgware.SR2.

**Comment by ryanjbaxter** [comment]
I dont see anything in the stacktrace that has to do with Spring Cloud.  If you can prove some code in Spring Cloud Commons is causing the issue then please open an issue here https://github.com/spring-cloud/spring-cloud-commons

**Comment by AlexandraMDan** [comment]
my issue appeared when I have updated the Spring Cloud config  version from Edgware.RELEASE to  Edgware.SR2. While investigating the issue, I noticed that it works if I override the version of spring-cloud-commons-dependencies used by Spring Cloud Config Edgware.SR2 and force the application to use spring-cloud-commons-dependencies version 1.3.0.RELEASE.
Should I open the issue to spring-cloud/spring-cloud-commons in this case?
